diff -urN co.orig/comodules/acl/acl_init.t co/comodules/acl/acl_init.t
--- co.orig/comodules/acl/acl_init.t	Tue Apr 15 14:46:07 2003
+++ co/comodules/acl/acl_init.t	Thu May 20 12:52:59 2004
@@ -7,7 +7,7 @@
   %%
 
 #if 0
-  ##creating acls for classes in meta_class and all objects but not for acl
+  // creating acls for classes in meta_class and all objects but not for acl
 
   my(@area) = ('sec', 'sys', 'usr' );
 
@@ -45,7 +45,7 @@
 			  where r.name = 'root'
 		    ");
 
-  ##creating acls for server_event scripts
+  // creating acls for server_event scripts
 
   $result = $conn->exec(" insert into co_sys_acl(id_permission,id_object,id_role,area,class_name)
 			  select
diff -urN co.orig/comodules/acl_class/acl_class_init.t co/comodules/acl_class/acl_class_init.t
--- co.orig/comodules/acl_class/acl_class_init.t	Tue Apr 15 14:46:07 2003
+++ co/comodules/acl_class/acl_class_init.t	Thu May 20 12:52:29 2004
@@ -5,7 +5,7 @@
  
   escape_perl
   %%
-  ##creating acl_class for all classes in meta_class and for role root
+  // creating acl_class for all classes in meta_class and for role root
 
   $result = $conn->exec(" insert into co_sys_acl_class 
                           select NEXTVAL('co_sys_acl_class_sqid'), 0, 0, p.id, r.id, 'sys', c.class_name 
